app/scripts/modules/core/widgets/spelText/spelAutocomplete.service.js

```diff
@@ -0,0 +1,327 @@
+'use strict';
+
+
+let angular = require('angular');
+
+module.exports = angular
+  .module('spinnaker.core.widget.spelAutocomplete', [
+    require('./jsonListBuilder'),
+    require('../../delivery/service/execution.service'),
+  ])
+  .factory('spelAutocomplete', function($q, jsonListBuilder, executionService) {
+    let brackets = [{open: '(', close: ')'}, {open: '[', close: ']'}, ];
+    let quotes = [{open: '\'', close: '\'' }, {open: '\"', close: '\"' }];
+    let helperFunctions = ['alphanumerical', 'fromJson', 'fromUrl', 'jsonFromUrl', 'judgement', 'stage', 'toBoolean', 'toFloat', 'toInt', 'toJson'];
+    let helperParams = ['execution', 'parameters', 'trigger', 'scmInfo', 'scmInfo.sha1', 'scmInfo.branch', 'deployedServerGroups'];
```

Comment:
can we restrict scmInfo to only Jenkins triggers? doesn't work with anything else, really